src/lxml/serializer.pxi

```diff
@@ -954,7 +954,15 @@ cdef class _IncrementalFileWriter:
                     if self._status > WRITER_IN_ELEMENT or content.strip():
                         raise LxmlSyntaxError("not in an element")
                 content = _utf8(content)
-                tree.xmlOutputBufferWriteEscape(self._c_out, _xcstr(content), NULL)
+                if c_method == OUTPUT_METHOD_XML:
```

Comment:
This seems reversed. Shouldn't your special casing apply *only* to HTML, instead of everything but XML?